lib/async/http/endpoint.rb

```diff
@@ -18,16 +18,26 @@ module Async
 	module HTTP
 		# Represents a way to connect to a remote HTTP server.
 		class Endpoint < ::IO::Endpoint::Generic
+			SCHEMES = ['HTTP', 'HTTPS', 'WS', 'WSS'].to_h do |scheme|
+				[scheme.downcase, URI.scheme_list[scheme]]
+			end
+			
 			def self.parse(string, endpoint = nil, **options)
 				url = URI.parse(string).normalize
 				
 				return self.new(url, endpoint, **options)
 			end
 			
 			# Construct an endpoint with a specified scheme, hostname, optional path, and options.
+			#
+			# @parameter scheme [String] The scheme to use, e.g. "http" or "https".
+			# @parameter hostname [String] The hostname to connect to (or bind to).
+			# @parameter *options [Hash] Additional options, passed to {#initialize}.
 			def self.for(scheme, hostname, path = "/", **options)
 				# TODO: Consider using URI.for once it becomes available:
-				uri_klass = URI.scheme_list[scheme.upcase] || URI::HTTP
+				uri_klass = SCHEMES.fetch(scheme.downcase) do
+					raise ArgumentError, "Unsupported scheme: #{scheme}"
```

Comment:
Not sure how you prefer to format your exception messages, but I always add `.inspect` to any variables so I can identify non-String values that convert down to a String and look the same (ex: `Unsupported scheme: http` vs. `Unsupported scheme: :http`).